lib/full-site-editing/block-templates.php

```diff
@@ -93,18 +95,42 @@ function _gutenberg_get_template_files( $template_type ) {
 				// Subtract ending '.html'.
 				-5
 			);
-			$template_files[] = array(
-				'slug'  => $template_slug,
-				'path'  => $template_file,
-				'theme' => $theme_slug,
-				'type'  => $template_type,
+			$template_files[] = _gutenberg_add_template_part_section_info(
+				array(
+					'slug'  => $template_slug,
+					'path'  => $template_file,
+					'theme' => $theme_slug,
+					'type'  => $template_type,
+				)
 			);
 		}
 	}
 
 	return $template_files;
 }
 
+/**
+ * Attempts to add the template part's section information from the
+ * experimental-theme.json file to the input array.
+ *
+ * @param array $template_info Template to add information to (requires 'type' and 'slug' fields).
+ *
+ * @return array Template.
+ */
+function _gutenberg_add_template_part_section_info( $template_info ) {
```

Comment:
`Non-blocking:`

The name of this method confused me a little. When I first saw it on l52 and l98, I thought that we were _always_ appending template part section info. In reality, we add it conditionally.  

Curious how we would feel about breaking down `_gutenberg_add_template_part_section_info` into two methods with distinct responsibilities and refactoring like so:

```php
function _gutenberg_get_template_file( $template_type, $slug ) {
                         // only included relevant code in this method
			$template_info = array(
				'slug'  => $template_slug,
				'path'  => $template_file,
				'theme' => $theme_slug,
				'type'  => $template_type,
			)

			if ( _should_add_section_info( $template_type )) {
				$template_files[] = _gutenberg_add_template_part_section_info($template_info);
			} else {
				$template_files[] = $template_info;
			}
		}
	}

	return $template_files;
}

function _should_add_section_info( $template_type ) {
	return  'wp_template_part' === $template_type && is_readable( locate_template( 'experimental-theme.json' ) )
};

function _gutenberg_add_template_part_section_info( $template_info ) {
	$theme_json = file_get_contents( locate_template( 'experimental-theme.json' ) );
	$data       = json_decode(
		$theme_json,
		true
	);
	if ( isset( $data['template-parts'][ $template_info['slug'] ]['section'] ) ) {
		$template_info['section'] = gutenberg_filter_template_part_section_type( $data['template-parts'][ $template_info['slug'] ]['section'] );
	}

	return $template_info;
}
```

Reply:
I was scratching my head over the best way to go about this as well.  That may make sense!  Initially, I wanted to avoid having to put the same conditional logic for whether or not to add the section info in both `_gutenberg_get_template_file` and `gutenberg_get_template_files` which is why I moved it all into that add function. 


Refactoring it like this means we will have that same if/else in both single/plural versions of the get functions, but thats not necessarily the end of the world... I just noticed myself typing the same thing in 2 places and wanted to consolidate it.

Do you think it makes more sense to separate this into 2 functions, or to rename the function to `*_conditionally_add_*`? 🤷‍♀️

Reply:
> Refactoring it like this means we will have that same if/else in both single/plural versions of the get functions, but thats not necessarily the end of the world... I just noticed myself typing the same thing in 2 places and wanted to consolidate it.

Ah that makes sense.

> Do you think it makes more sense to separate this into 2 functions, or to rename the function to *_conditionally_add_*? 🤷‍♀️

Hmm...I'm thinking about other names as well. I do think it would be more informative than what we currently have.

> I wanted to avoid having to put the same conditional logic for whether or not to add the section info in both _gutenberg_get_template_file and gutenberg_get_template_files which is why I moved it all into that add function.
One other option is to extract the conditional logic in both 

One other option is extracting the similar conditional logic into its own method as well 🤔

```php
function _gutenberg_get_template_file( $template_type, $slug ) {
                         // only included relevant code in this method
			$template_info = array(
				'slug'  => $template_slug,
				'path'  => $template_file,
				'theme' => $theme_slug,
				'type'  => $template_type,
			)

                        $template_files[] = _generate_template_file_from_info($template_info);
		}
	}

	return $template_files;
}

function _gutenberg_get_template_files( $template_type ) {
                         // only included relevant code in this method
			$template_info = array(
				'slug'  => $template_slug,
				'path'  => $template_file,
				'theme' => $theme_slug,
				'type'  => $template_type,
			)

                        $template_files[] = _generate_template_file_from_info($template_info);
		}
	}

	return $template_files;
}
```

I think renaming it _or_ creating an additional method to DRY out conditional logic would be good

Reply:
Im not totally following the last example.  It looks like what is happening now with just the function renamed (same input, used in same place.)?

Reply:
I updated the last example to include more information. I should've clarified a few things:

> I wanted to avoid having to put the same conditional logic for whether or not to add the section info in both _gutenberg_get_template_file and gutenberg_get_template_files which is why I moved it all into that add function.

The last example builds off of the first example. In the first example, we'd end up with duplicated conditional logic in `_gutenberg_get_template_file` and `gutenberg_get_template_files`. In the last example, we're adding one additional method called `_generate_template_file_from_info` to extract the duplicated conditional logic into its own method. Does that make more sense?

Reply:
Oh I see what you're saying! Yes the solution ends up matching the structure of what you originally had, except with a few more declarative helper methods.

Reply:
Gotcha, thanks for clarifying.  I see what you are saying.  This would make sense to me if these helper methods had a use elsewhere, but Im not sure its worth splitting up into 3 functions for short helper methods that will only be used in 1 place.  With that in mind id be more inclined to just rename the original function with `*_conditionally_*`.

Reply:
b1138e0 lgtm!